src/NuGet.Server/Infrastructure/ServerPackageRepository.cs

```diff
@@ -458,17 +458,25 @@ private HashSet<ServerPackage> ReadPackagesFromDisk()
 
                 Parallel.ForEach(packages, package =>
                 {
-                    // Create server package
-                    var serverPackage = CreateServerPackage(package, enableDelisting);
-
-                    // Add the package to the cache, it should not exist already
-                    if (cachedPackages.Contains(serverPackage))
+                    try
                     {
-                        _logger.Log(LogLevel.Warning, "Duplicate package found - {0} {1}", package.Id, package.Version);
+                        // Create server package
+                        var serverPackage = CreateServerPackage(package, enableDelisting);
+
+                        // Add the package to the cache, it should not exist already
```

Comment:
Why is the try/catch wrapping the cache code? It should probably be just around the create server package. And even better we can make it in CreateServerPackage method itself (so reusable for future callers), and make that method return a bool.

thoughts?